packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -170,12 +168,12 @@ export class EngineReportingExtension<TContext = any>
     // While clientAddress could be a part of the protobuf, we'll ignore it for
     // now, since the backend does not group by it and Engine frontend will not
     // support it in the short term
-    const { clientName, clientVersion } = this.generateClientInfo({
-      context: o.context,
-      extensions: o.extensions,
-    });
-    this.trace.clientName = clientName || '';
+    const { clientName, clientVersion, clientId } = this.generateClientInfo(
+      o.requestContext,
+    );
     this.trace.clientVersion = clientVersion || '';
+    this.trace.clientId = clientId || clientName || '';
+    this.trace.clientName = clientName || clientId || '';
```

Comment:
These fallbacks were a bit unexpected and seem to make things more difficult to document. When my client name is also my client id, I would have trouble figuring out the difference and what to set each to. Probably can be solved with documentation, so I'll look out for the PR that adds it in.

Reply:
@evans You’re absolutely right. I hope we can find an approach that works, soon, that will avoid the need for additional documentation.

Reply:
@evans @jbaxleyiii @abernix the BE code already does this exact thing automatically.